src/eos-crash-metrics.c

```diff
@@ -137,6 +145,111 @@ normalize_path (char *path)
     path[i] = path[i] == '!' ? '/' : path[i];
 }
 
+static GPtrArray *
+get_installed_ref_for_executable (FlatpakInstallation *installation,
+                                  const char *executable_name,
+                                  GError **error)
+{
+  g_autoptr(GPtrArray) xrefs = NULL;
+  GPtrArray *applications = NULL;
+  FlatpakInstalledRef *xref = NULL;
+  guint i;
+
+  /* get apps and runtimes */
+  xrefs = flatpak_installation_list_installed_refs (installation,
+                                                    NULL, error);
+  if (xrefs == NULL)
+    return NULL;
+
+  applications = g_ptr_array_new ();
+
+  for (i = 0; i < xrefs->len; i++)
+    {
+      g_autofree gchar *executable_path = NULL;
+      FlatpakInstalledRef *xref = g_ptr_array_index (xrefs, i);
+
+      /* only apps */
+      if (flatpak_ref_get_kind (FLATPAK_REF (xref)) != FLATPAK_REF_KIND_APP)
+        continue;
+
+      executable_path = g_build_filename (flatpak_installed_ref_get_deploy_dir (FLATPAK_INSTALLED_REF (xref)),
+                                          "files",
+                                          "bin",
+                                          executable_name,
+                                          NULL);
+      /* found a Flatpak with the same application name */
+      if (g_file_test (executable_path, G_FILE_TEST_EXISTS))
+        g_ptr_array_add (applications, g_object_ref (xref));
+    }
+
+  return applications;
+}
+
+static FlatpakInstalledRef *
+get_flatpak_installed_ref (const char *path, GError **error)
+{
+  g_autofree char *executable_name = NULL;
+  g_autoptr(GPtrArray) applications = NULL;
+  g_autoptr(FlatpakInstallation) installation = flatpak_installation_new_system (NULL, error);
+
+  if (installation == NULL)
+    return NULL;
+
+  executable_name = g_path_get_basename (path);
+
+  applications = get_installed_ref_for_executable (installation, executable_name, error);
+  if (!applications)
+    return NULL;
+
+  /* Even if we find Flatpaks with the same application name
+     we fallback to the first one we find for now */
+  FlatpakInstalledRef *xref = g_ptr_array_index (applications, 0);
+  return xref;
+}
+
+static inline void
+sanitize_name (gchar *name)
+{
+  gchar *tmp = strchr (name, '/');
+
+  if (tmp != NULL)
+    *tmp = '\0';
+}
+
+static const char *
+get_associated_runtime (FlatpakInstalledRef *ref)
+{
+  g_autoptr(GBytes) metadata = NULL;
+  gchar *sdk = NULL;
+  g_autoptr(GKeyFile) key_file = NULL;
+  const gchar *metadata_data;
+  gsize metadata_len;
+  g_autoptr(GError) local_error = NULL;
+
+  metadata = flatpak_installed_ref_load_metadata (FLATPAK_INSTALLED_REF (ref), NULL, &local_error);
+
+  if (metadata == NULL)
+    {
+      g_warning ("%s", local_error->message);
```

Comment:
You could have a slightly more explanatory message here (and below)

Reply:
Perhaps prefer to bubble up the error